operators/pkg/controller/elasticsearch/driver/version_wide_resources.go

```diff
@@ -25,66 +21,52 @@ import (
-	ExtraFilesSecret corev1.Secret
-	// GenericUnecryptedConfigurationFiles contains non-secret files Pods with this version should have access to.
-	GenericUnecryptedConfigurationFiles corev1.ConfigMap
+	// ClusterSecrets contains possible user-defined secret files we want to have access to in the containers.
+	ClusterSecrets corev1.Secret
+	// GenericUnencryptedConfigurationFiles contains non-secret files Pods with this version should have access to.
+	GenericUnencryptedConfigurationFiles corev1.ConfigMap
 }
 
 func reconcileVersionWideResources(
 	c k8s.Client,
 	scheme *runtime.Scheme,
 	es v1alpha1.Elasticsearch,
-	trustRelationships []v1alpha1.TrustRelationship,
 ) (*VersionWideResources, error) {
 	expectedConfigMap := configmap.NewConfigMapWithData(k8s.ExtractNamespacedName(&es), settings.DefaultConfigMapData)
 	err := configmap.ReconcileConfigMap(c, scheme, es, expectedConfigMap)
 	if err != nil {
 		return nil, err
 	}
 
-	trustRootCfg := nodecerts.NewTrustRootConfig(es.Name, es.Namespace)
-
-	// include the trust restrictions from the trust relationships into the trust restrictions config
-	for _, trustRelationship := range trustRelationships {
-		trustRootCfg.Include(trustRelationship.Spec.TrustRestrictions)
-	}
-
-	trustRootCfgData, err := json.Marshal(&trustRootCfg)
-	if err != nil {
-		return nil, err
-	}
-
-	expectedExtraFilesSecret := corev1.Secret{
+	// TODO: this may not exactly fit the bill of being specific to a version
+	expectedClusterSecretsSecret := corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: es.Namespace,
-			Name:      name.ExtraFilesSecret(es.Name),
-		},
-		Data: map[string][]byte{
-			nodecerts.TrustRestrictionsFilename: trustRootCfgData,
+			Name:      name.ClusterSecretsSecret(es.Name),
 		},
+		Data: map[string][]byte{},
 	}
 
-	var reconciledExtraFilesSecret corev1.Secret
+	var reconciledClusterSecretsSecret corev1.Secret
 
 	if err := reconciler.ReconcileResource(reconciler.Params{
 		Client:     c,
 		Scheme:     scheme,
 		Owner:      &es,
-		Expected:   &expectedExtraFilesSecret,
-		Reconciled: &reconciledExtraFilesSecret,
+		Expected:   &expectedClusterSecretsSecret,
+		Reconciled: &reconciledClusterSecretsSecret,
 		NeedsUpdate: func() bool {
 			// .Data might be nil in the secret, so make sure to initialize it
-			if reconciledExtraFilesSecret.Data == nil {
-				reconciledExtraFilesSecret.Data = make(map[string][]byte, 1)
+			if reconciledClusterSecretsSecret.Data == nil {
+				reconciledClusterSecretsSecret.Data = make(map[string][]byte, 0)
 			}
-			currentTrustConfig, ok := reconciledExtraFilesSecret.Data[nodecerts.TrustRestrictionsFilename]
 
-			return !ok || !bytes.Equal(currentTrustConfig, trustRootCfgData)
+			// TODO: compare items that we may want to reconcile here
+
+			return false
 		},
 		UpdateReconciled: func() {
-			reconciledExtraFilesSecret.Data[nodecerts.TrustRestrictionsFilename] = trustRootCfgData
+			// TODO: add items to reconcile here
```

Comment:
IIUC: for now, this is an empty secret placeholder with no content?

Reply:
Correct. The end result may be that we discard this, but that decision probably naturally happens post #776.